openpgp/packet/signature.go

```diff
@@ -842,18 +1045,41 @@ func (sig *Signature) Serialize(w io.Writer) (err error) {
 }
 
 func (sig *Signature) serializeBody(w io.Writer) (err error) {
-	hashedSubpacketsLen := uint16(uint16(sig.HashSuffix[4])<<8) | uint16(sig.HashSuffix[5])
-	fields := sig.HashSuffix[:6+hashedSubpacketsLen]
+	var fields []byte
+	if sig.Version == 6 {
+		// v6 signatures use 4 ocets for length
+		hashedSubpacketsLen := uint32(uint32(sig.HashSuffix[4])<<24) | 
+							   uint32(uint32(sig.HashSuffix[5])<<16) | 
+							   uint32(uint32(sig.HashSuffix[6])<<8)  | 
+							   uint32(sig.HashSuffix[7])
```

Comment:
Also here no tabs for alignment please, as tab size is not consistent. You could do something like
```suggestion
		hashedSubpacketsLen :=
			uint32(uint32(sig.HashSuffix[4])<<24) | 
			uint32(uint32(sig.HashSuffix[5])<<16) | 
			uint32(uint32(sig.HashSuffix[6])<<8)  | 
			uint32(sig.HashSuffix[7])
```
or whatever gofmt comes up with. Similarly below